[Term Entry] Java Queue: clear() (#7541)
Codecademy/codepedia-content: content/java/concepts/queue/terms/clear/clear.md

import java.util.LinkedList;
import java.util.Queue;

public class QueueClearBasic {
  public static void main(String[] args) {
    // Create a queue using LinkedList implementation
    Queue<String> queue = new LinkedList<>();

    // Add elements to the queue
    queue.offer("First");
    queue.offer("Second");
    queue.offer("Third");
    queue.offer("Fourth");

    // Display the original queue
    System.out.println("Original queue: " + queue);
    System.out.println("Queue size before clear: " + queue.size());

    // Clear all elements from the queue
    queue.clear();

    // Display the queue after clearing
    System.out.println("Queue after clear: " + queue);
    System.out.println("Queue size after clear: " + queue.size());
  }
}
```

The output of this code is:

```shell
Original queue: [First, Second, Third, Fourth]
Queue size before clear: 4
Queue after clear: []
Queue size after clear: 0
```

This example creates a queue, adds four string elements, displays the original content, clears the queue, and then shows the empty result. The `clear()` method removes all elements but keeps the queue structure intact for future use.

## Example 2: Task Processing System

> This example shows how `clear()` can be used in a task processing system where you need to reset the task queue after completing a batch of operations:

Reviewer: ```suggestion
This example creates a queue, adds four string elements, displays the original content, clears the queue, and then shows the empty result. The `clear()` method removes all elements but keeps the queue structure intact for future use.

## Example 2: Task Processing System

This example illustrates how `clear()` can be utilized in a task processing system to reset the task queue after completing a batch of operations.
```

    while (!tempQueue.isEmpty()) {
      System.out.println("Accessing cache entry: " + tempQueue.poll());
    }

    // Check memory usage and clear cache if needed
    boolean memoryThresholdExceeded = true; // Simulated condition

    if (memoryThresholdExceeded) {
      System.out.println("\nMemory threshold exceeded. Clearing cache...");
      cacheQueue.clear();
      System.out.println("Cache cleared successfully.");
      System.out.println("Current cache size: " + cacheQueue.size());
      System.out.println("Cache is empty: " + cacheQueue.isEmpty());

      // Cache is now ready for new entries
      System.out.println("\nCache system ready for new entries.");
    }
  }
}
```

The output of this code is:

```shell
Cache entries (priority order): [3, 5, 15, 10, 12]
Cache size: 5
Highest priority item: 3

Processing cache entries:
Accessing cache entry: 3
Accessing cache entry: 5
Accessing cache entry: 10
Accessing cache entry: 12
Accessing cache entry: 15

Memory threshold exceeded. Clearing cache...
Cache cleared successfully.
Current cache size: 0
Cache is empty: true

Cache system ready for new entries.
```

> This example shows how `clear()` method works with different Queue implementations like [PriorityQueue](https://www.codecademy.com/resources/docs/java/priorityqueue). The method is particularly useful in cache management systems where you need to periodically clear cached data to free up memory.

Reviewer: ```suggestion
The following text explains how the `clear()` method works with different Queue implementations, such as [PriorityQueue](https://www.codecademy.com/resources/docs/java/priorityqueue). This method is particularly useful in cache management systems where there is a need to periodically clear cached data to free up memory.
```
